### PR TITLE
Migrate roi_align and roi_align_grad to phi. test=kunlun

### PR DESCRIPTION
**Pull request by ZibinGuo**
### PR types
Others

### PR changes
OPs

### Describe
Migrate roi_align and roi_align_grad to phi. test=kunlun

### CODE REVIEW

**Comment by ZibinGuo** [comment]
您的邮件已收到，我将尽快回信。谢谢您！Your email has been received and I will reply as soon as possible. Thank you！

File: paddle/phi/kernels/xpu/roi_align_grad_kernel.cc [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                                    height,
+                                    width,
+                                    out_grad.dims()[0],
+                                    pooled_height,
+                                    pooled_width,
+                                    spatial_scale,
+                                    sampling_ratio,
+                                    true,
+                                    aligned);
+    PADDLE_ENFORCE_EQ(
```
Comment (houj04) [inline paddle/phi/kernels/xpu/roi_align_grad_kernel.cc]: 用PADDLE_ENFORCE_XDNN_SUCCESS那个宏

File: paddle/phi/kernels/xpu/roi_align_kernel.cc [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+                             height,
+                             width,
+                             out->dims()[0],
+                             pooled_height,
+                             pooled_width,
+                             spatial_scale,
+                             sampling_ratio,
+                             true,
+                             aligned);
+  PADDLE_ENFORCE_EQ(
```
Comment (houj04) [inline paddle/phi/kernels/xpu/roi_align_kernel.cc]: 同上

**Review by houj04 — DISMISSED** [review]

**Review by YuanRisheng — DISMISSED** [review]

File: paddle/phi/kernels/xpu/roi_align_grad_kernel.cc (hunk 1 above)
Comment (ZibinGuo) [reply paddle/phi/kernels/xpu/roi_align_grad_kernel.cc]: 已修改

File: paddle/phi/kernels/xpu/roi_align_kernel.cc (hunk 2 above)
Comment (ZibinGuo) [reply paddle/phi/kernels/xpu/roi_align_kernel.cc]: 已修改

File: paddle/phi/kernels/xpu/roi_align_kernel.cc [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                             height,
+                             width,
+                             out->dims()[0],
+                             pooled_height,
+                             pooled_width,
+                             spatial_scale,
+                             sampling_ratio,
+                             true,
+                             aligned);
+  PADDLE_ENFORCE_XDNN_SUCCESS(r, "roi_align_grad");
```
Comment (houj04) [inline paddle/phi/kernels/xpu/roi_align_kernel.cc:140]: 这里有typo，攒着以后改吧，节约时间&&CI资源

**Review by houj04 — APPROVED** [review]

**Review by YuanRisheng — APPROVED** [review]

**Pull request merged**